OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -97,11 +97,10 @@ public override bool Tick(Actor self)
 					QueueChild(new Land(self, Target.FromActor(cargo), -carryableBody.LocalToWorld(localOffset), carryableFacing.Facing));
 
 					// Pause briefly before attachment for visual effect
-					if (delay > 0)
```

Comment:
Is there a benefit to queueing a `Wait` activity with an zero or negative wait time, or do you just think this is negligible? (Doesn't this in theory make it wait one tick longer if the delay is 0, and infinite if delay < 0, although that case is unlikely?)

Reply:
I just think it is cleaner if `Wait` does the check and shortcuts on failure instead of every activity that calls `Wait` having to check for it. If the activity returns `true` on the first tick `RunActivity` goes to the next activity immediately so it wouldn't lose a tick.

But, looking more closely, it seems that `Wait` doesn't currently handle this properly. However, there are already some instances where we call `Wait` as a child without doing any checks. So, I'll revert the changes here and file a new PR to add the necessary checks to `Wait`.

Reply:
Ok, that sounds reasonable. Imho you don't have to revert this here if you want to fix `Wait` in a future PR anyway.